### ISSUE TITLE
Buttons could not be interacted with while dragging window on Windows

### ISSUE THREAD

**Issue by feelingnothing** [emilk/egui#1245]
**Screenshots**
![result](https://user-images.githubusercontent.com/47575622/152663386-8b54ed9a-5166-44c6-888b-bdddb330114a.gif)

**Desktop (please complete the following information):**
 - OS: Windows 10 21H1

**Additional context**
Duplicate of https://github.com/emilk/egui/issues/1215 , because I can't reopen issue that was closed not by myself
```rust
if egui::TopBottomPanel::new(TopBottomSide::Top, "title").show(&ctx, |ui| {
    ui.horizontal(|ui| {
        ui.label("test");
        ui.with_layout(Layout::right_to_left(), |ui| {
            if ui.button("X").interact(Sense::click()).clicked() {
                std::process::exit(0);
            };
        })
    });
}).response.interact(Sense::click_and_drag()).dragged() {
    _frame.drag_window();
};
egui::CentralPanel::default().show(&ctx, |ui| {
    if ui.button("text2").clicked() {
        println!("123");
    }
});
```

**Comment by feelingnothing** [comment]
Some debugging after I found that we do not get events while dragging on windows, could be winit issue, going to continue debugging anyway

**Comment by feelingnothing** [comment]
> Is this fixed by #1108 ?

This PR is quite old, so I guess I tested with it already, there is a PR that I created for `winit`: https://github.com/rust-windowing/winit/pull/2200
Waiting on this to be pulled and I have to make a PR for dependencies of `egui` to include the fix.

**Comment by feelingnothing** [comment]
Should be fixed in winit ">=0.28", which fits the current version that "egui-winit" is utilizing, if someone could test it once and for all I would close the issue.

**Comment by urholaukkarinen** [comment]
This also happens on Linux (X11 at least).
Somewhat related winit issue: https://github.com/rust-windowing/winit/issues/2413

**Comment by sigh-gone** [comment]
Also happens on MacOS Ventura 13.5.2
